### PR TITLE
introduce a feature to select the correct metadata for either peregrine or spiritnet

### PR DESCRIPTION
**Pull request by trusch**
introduce a feature to select the correct metadata for either peregrine or spiritnet
By default it uses the spiritnet metadata, but this is in preparation for the :peregrine and :spiritnet tags

Actually it doesnt make a difference, but perhaps in the future peregrine and spiritnet would diverge too much. By now you can compile for peregrine and execute against spiritnet and vica verca without problems.

## fixes KILTProtocol/ticket#312
Please include a summary of the changes provided with this pull request and which issue has been fixed. 
Please also provide some context if necessary.

## How to test:
Please provide a brief step-by-step instruction.
If necessary provide information about dependencies (specific configuration, branches, database dumps, etc.)

- Step 1
- Step 2
- etc.

## Checklist:

- [ ] I have verified that the code works
- [ ] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [ ] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [ ] I have documented the changes (where applicable)
    * Either PR or Ticket to update [the Docs](https://github.com/KILTprotocol/docs)
    * Link the PR/Ticket here

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -38,3 +38,7 @@ rhai = { version = "1.15.1", features = ["serde", "sync"] }
 once_cell = "1.18.0"
 base64 = "0.21.2"
 anyhow = "1.0.75"
+
+[features]
+default = []
```
Comment (ntn-x2) [inline Cargo.toml:43]: Should it rather have a `spiritnet` feature, with the `default` feature not using either of them by default? Or even removing the `default` feature for now, and only explicitly have `spiritnet`  and `peregrine` features?

Comment (trusch) [reply Cargo.toml:43]: having one feature to activate "test environment" mode is simpler in my opinion.

File: metadata-peregrine.scale
Comment (Ad96el) [inline metadata-peregrine.scale:1]: I think it is useful to have the version in the file name. It makes it more clear when to update the scale file after a chain upgrade.

File: Cargo.toml (hunk 1 above)
Comment (ntn-x2) [reply Cargo.toml:43]: The change would reflect the logic in the Docker image naming. I am also fine with this. Your call as to whether accept or ignore my suggestion 👍

Comment (trusch) [reply Cargo.toml:43]: Can you tell me how to do mutually exclusive feature flags? I know I could have something like "if(spiritnet) and not peregrine" but this wouldnt prevent the user trying to build something with spiritnet and peregrine enabled. If we just have one feature flag its easier. I would be honestly interested to know a good pattern to build this, but wasnt able to find something online.

**Review by Ad96el — APPROVED** [review]
Consider removing the default feature flag in Cargo.toml as it doesn't serve a significant purpose. Having only the peregrine features should be sufficient, in my opinion. The decision is up 2 u

**Pull request merged**